plugins/catalog-backend/src/database/DatabaseManager.ts

```diff
@@ -44,26 +70,45 @@ export class DatabaseManager {
         );
 
         const readerOutput = await reader.read(location.type, location.target);
+
         for (const readerItem of readerOutput) {
           if (readerItem.type === 'error') {
             logger.debug(readerItem.error);
             continue;
           }
-
-          const parserOutput = await parser.parse(readerItem.data);
-          if (parserOutput.kind === 'Component') {
-            const component = parserOutput.component;
-            const dbc: AddDatabaseComponent = {
-              locationId: location.id,
-              name: component.metadata.name,
-            };
-            await database.addOrUpdateComponent(dbc);
+          let parserOutput;
+          try {
+            parserOutput = await parser.parse(readerItem.data);
+            if (parserOutput.kind === 'Component') {
+              const component = parserOutput.component;
+              const dbc: AddDatabaseComponent = {
+                locationId: location.id,
+                name: component.metadata.name,
+              };
+              await database.addOrUpdateComponent(dbc);
+              await DatabaseManager.logUpdateSuccess(
```

Comment:
would be nice to run these two in a transaction

Reply:
I considered implementing a transaction here but I thought that you might want to discuss different approaches beforehand (a way to pass knex builder/transaction to each function that queries db). How about creating a separate issue for that?